package.json

```diff
@@ -104,6 +105,9 @@
     "src/**/*.{js,ts,jsx,tsx}": [
       "eslint --fix ./src",
       "prettier --write"
+    ],
+    "src/**/*.{ts,tsx}": [
+      "tsc-files --noEmit src/react-app-env.d.ts src/typings.d.ts"
```

Comment:
mmm how does that work? it runs check for 2 files only?

Reply:
No :) No) lint-staged passes all changed files to this command, the first two are global type declarations, which must be included in every run

After your comment I realized that lint-staged is not suitable because it only checks changed files and may miss errors in dependent files that have not been changed, I added checking the entire project in each commit, this is longer but more correct